openwisp_utils/tests.py

```diff
@@ -76,3 +77,67 @@ def stopTestRun(self):
 class TimeLoggingTestRunner(DiscoverRunner):
     def get_resultclass(self):
         return TimeLoggingTestResult
+
+
+class RedirectOutput(object):
```

Comment:
Thing this can be shortened:

```
def __call__(self, function):
        def wrapped_function(*args, **kwargs):

            if self.stdout:
                sys.stdout = self.stdout
            if self.stderr:
                sys.stderr = self.stderr

                try:
                    if self.stdout and self.stderr:
                         function(*args, self.stdout, self.stderr, **kwargs)
                    elif self.stdout:
                         function(*args, self.stdout, **kwargs)
                    else:
                         function(*args, self.stderr, **kwargs)
                except Exception:
                    function(*args, **kwargs)
                finally:
                    self.stdout.close()
                    if self.stdout:
                         sys.stdout = sys.__stdout__
                    if self.stderr:
                         sys.stderr = sys.__stderr__

        return wrapped_function

```

Reply:
Thanks @atb00ker for your guidance.